core/src/test/java/com/scottlogic/datahelix/generator/core/walker/decisionbased/RowSpecTreeSolverTests.java

```diff
@@ -108,6 +116,138 @@ void createRowSpecs_whenRootNodeHasSomeDecisions_returnsRowSpecOfRoot() {
+        Map<Field, FieldSpec> option0 = new HashMap<>();
+        option0.put(fieldA, FieldSpecFactory.fromType(fieldA.getType()));
+        option0.put(fieldB, FieldSpecFactory.fromList(distributedListOfOneItem("1", 0.25)));
+        expectedRowSpecs.add(new WeightedElement<>(new RowSpec(fields, option0, Collections.emptyList()), 0.25));
+        Map<Field, FieldSpec> option1 = new HashMap<>();
+        option1.put(fieldA, FieldSpecFactory.fromType(fieldA.getType()));
+        option1.put(fieldB, FieldSpecFactory.fromList(distributedListOfOneItem("2", 0.75)));
+        expectedRowSpecs.add(new WeightedElement<>(new RowSpec(fields, option1, Collections.emptyList()), 0.75));
+
+        assertThat(
+            rowSpecs,
+            sameBeanAs(expectedRowSpecs));
+    }
+
+    @Test
+    void createRowSpecs_whenRootNodeHasSomeWeightedDecisionsAndAllValuesAreExcluded_returnsCorrectlyWeightedRowSpecs() {
+        //Arrange
+        ConstraintNode root = TestConstraintNodeBuilder.constraintNode()
+            .where(fieldB).isInSet(
+                new InSetRecord("1", 0.25d),
+                new InSetRecord("2", 0.75d))
+            .withDecision(
+                TestConstraintNodeBuilder.constraintNode()
+                    .where(fieldB).isInSet("3"),
+                TestConstraintNodeBuilder.constraintNode()
+                    .where(fieldB).isInSet("4"))
+            .build();
+        DecisionTree tree = new DecisionTree(root, fields);
+
+        //Act
+        Set<WeightedElement<RowSpec>> rowSpecs = rowSpecTreeSolver.createRowSpecs(tree).collect(Collectors.toSet());
+
+        //Assert
+        assertThat(
+            rowSpecs.isEmpty(),
+            is(true));
+    }
+
+    @Test
+    void createRowSpecs_whenRootNodeHasUnweightedDecisionsAndAllValuesAreAllowed_returnsCorrectlyWeightedRowSpecs() {
+        //Arrange
+        ConstraintNode root = TestConstraintNodeBuilder.constraintNode()
+            .where(fieldB).isInSet("1", "2")
+            .withDecision(
+                TestConstraintNodeBuilder.constraintNode()
+                    .where(fieldB).isInSet("1", "2", "3"),
+                TestConstraintNodeBuilder.constraintNode()
+                    .where(fieldB).isInSet("1", "2", "4"))
+            .build();
+        DecisionTree tree = new DecisionTree(root, fields);
+
+        //Act
+        Set<WeightedElement<RowSpec>> rowSpecs = rowSpecTreeSolver.createRowSpecs(tree).collect(Collectors.toSet());
+
+        //Assert
+        Set<WeightedElement<RowSpec>> expectedRowSpecs = new HashSet<>();
+        Map<Field, FieldSpec> option0 = new HashMap<>();
+        option0.put(fieldA, FieldSpecFactory.fromType(fieldA.getType()));
+        option0.put(fieldB, FieldSpecFactory.fromLegalValuesList(Arrays.asList("1", "2")));
+        expectedRowSpecs.add(new WeightedElement<>(new RowSpec(fields, option0, Collections.emptyList()), 1));
+        Map<Field, FieldSpec> option1 = new HashMap<>();
```

Comment:
repeated same block of 4 lines.  Maybe just do option1 = new HashMap<>(option0) so just shallow copies it.

Reply:
I've added some helper functions to remove some of this repetition